### ISSUE TITLE
js_errors: true Doesn't Reraise Errors

### ISSUE THREAD

**Issue by factologist** [teampoltergeist/poltergeist#721]
Hullo! I'm attempting to get Poltergeist to fail tests when JS errors are present in the web console, but the `js_errors: true` boolean doesn't seem to be doing it for me. Note, the provided website in `test.rb` has JS Console errors on Chrome, but warnings on Safari. I *have* used it on internal sites that have JS errors on all browsers without the expected failures, though. Am I overlooking something completely?

There is not an assertion to check for anything on the page, because my assumption is that the `js_errors: true` would be enough to fail a test. Perhaps that assumption is incorrect, and an assertion of some kind needs to be added to fail a test due to JS console errors. Help appreciated!

`test.rb`:
```
require_relative '../spec_helper'

feature "JS Tests" do
  puts "JS Tests"

  scenario "website should be free of js errors", :js => true do
    visit 'http://isp.netscape.com/'
  end

end
```

`spec_helper.rb`:
```
require 'rubygems'
require 'bundler/setup'
require 'rspec'
require 'capybara/rspec'
require 'capybara/poltergeist'

Capybara.javascript_driver = :poltergeist

Capybara.register_driver :poltergeist do |app|
    options = {
              js_errors: true,
              timeout: 60,
              debug: false,
              phantomjs_options: ['--ignore-ssl-errors=yes', '--ssl-protocol=TLSv1']
              }
  Capybara::Poltergeist::Driver.new(app, options)
end
```

Behavior: test passes, without alerting / failing due to JS console errors
Expected: test would fail due to presence of JS console errors

Test output:

```
$ bundle exec rspec spec/test.rb 
JS Tests
.

Finished in 1.67 seconds (files took 1.14 seconds to load)
1 example, 0 failures
```

**Comment by twalpole** [comment]
js_errors: true should cause any action performed in the browser to raise an error if there are js_errors at the time the action returns.  It is possible the JS errors you're expecting are occurring after visit is returned, and would then require another action to be initiated on the page before you would get the error raised.  That being said - I don't see http://isp.netscape.com raising any js errors in my browser console when visiting it.

**Comment by factologist** [comment]
Thanks @twalpole .

> I don't see http://isp.netscape.com raising any js errors in my browser console when visiting it.

That is a bad example, then, sorry. For my purposes I'm using an internal page that can't be shared, but it's definitely got js errors on it.

> It is possible the JS errors you're expecting are occurring after visit is returned, and would then require another action to be initiated on the page before you would get the error raised.

I'll give this a shot, hopefully it's as simple as that. For my knowledge, it is *only* `JS errors that will trigger this failure, not network / request failures or CSS asset failures, etc. Correct?

**Comment by factologist** [comment]
I tried adding an assertion in the form of 

`click_button 'button'`. Solved it.

> It is possible the JS errors you're expecting are occurring after visit is returned, and would then require another action to be initiated on the page before you would get the error raised.

I think you were right, the error I was seeing (and hoping to fail as a result of seeing) comes from a slowly enacted post-load automatic action. Turns out Poltergeist is just super fast, and was giving the page the Thumbs Up before that action had finished and caused the error.

Thanks again!
